CloverConnectorImpl.js

```diff
@@ -143,12 +146,15 @@ CloverConnectorImpl.prototype.constructor = CloverConnectorImpl;
 
 /**
  * @param {Object} message - the message to send
+ * @param {boolean} handshaking - are we currently handshaking
  * @private
  */
-CloverConnectorImpl.prototype.sendMessage = function(message) {
+CloverConnectorImpl.prototype.sendMessage = function(message, handshaking) {
             try {
                 this.log.debug("sendMessage", message);
-                if(this.device == null) {
+                // If the device is null OR
+                // we are NOT handshaking, and we are not ready
+                if(this.device == null || (!handshaking && !this.isReady)) {
```

Comment:
not sure why handshaking matters here? shouldn't it be if(this.device == null || !this.isReady)?